src/main/java/software/amazon/encryption/s3/internal/BufferedCipherSubscriber.java

```diff
@@ -40,21 +36,22 @@ public class BufferedCipherSubscriber implements Subscriber<ByteBuffer> {
     private byte[] outputBuffer;
     private final Queue<ByteBuffer> buffers = new ConcurrentLinkedQueue<>();
 
-    BufferedCipherSubscriber(Subscriber<? super ByteBuffer> wrappedSubscriber, Long contentLength, CryptographicMaterials materials, byte[] iv) {
+    BufferedCipherSubscriber(Subscriber<? super ByteBuffer> wrappedSubscriber, Long contentLength, CryptographicMaterials materials, byte[] iv, long bufferSize) {
         this.wrappedSubscriber = wrappedSubscriber;
         if (contentLength == null) {
             throw new S3EncryptionClientException("contentLength cannot be null in buffered mode. To enable unbounded " +
                     "streaming, reconfigure the S3 Encryption Client with Delayed Authentication mode enabled.");
         }
-        if (contentLength > BUFFERED_MAX_CONTENT_LENGTH_BYTES) {
+        long bufferSizeInBytes = 1024 * 1024 * bufferSize;
+        if (contentLength > bufferSizeInBytes) {
             throw new S3EncryptionClientException(String.format("The object you are attempting to decrypt exceeds the maximum content " +
```

Comment:
nit: we can update this exception text to reflect the configurable nature of the buffer size e.g.:

```
"The object you are attempting to decrypt exceeds the maximum buffer size: " + bufferSize +
"for the default (buffered) mode. Either increase your buffer size when configuring your client, " + 
"or enable Delayed Authentication mode to disable buffered decryption."
```